presto-hive/src/test/java/com/facebook/presto/hive/TestS3SelectRecordCursor.java

```diff
@@ -0,0 +1,213 @@
+        buildSplitSchema(ddlSerializationValue, DEFAULT_TEST_COLUMNS);
+    }
+
+    @Test
+    public void shouldFilterColumnsWhichDoesNotMatchInTheHiveTable()
+    {
+        String ddlSerializationValue = "struct article { varchar address varchar company date date_pub int quantity}";
+        String expectedDDLSerialization = "struct article { date date_pub, int quantity}";
+        assertEquals(buildSplitSchema(ddlSerializationValue, DEFAULT_TEST_COLUMNS),
+                buildExpectedProperties(expectedDDLSerialization, DEFAULT_TEST_COLUMNS));
+    }
+
+    @Test
+    public void shouldReturnOnlyQuantityColumnInTheDDl()
+    {
+        String ddlSerializationValue = "struct article { varchar address varchar company date date_pub int quantity}";
+        String expectedDDLSerialization = "struct article { int quantity}";
+        assertEquals(buildSplitSchema(ddlSerializationValue, ARTICLE_COLUMN, QUANTITY_COLUMN),
+                buildExpectedProperties(expectedDDLSerialization, ARTICLE_COLUMN, QUANTITY_COLUMN));
+    }
+
+    @Test
+    public void shouldReturnProperties()
+    {
+        String ddlSerializationValue = "struct article { varchar article varchar author date date_pub int quantity}";
+        String expectedDDLSerialization = "struct article { varchar article, varchar author, date date_pub, int quantity}";
+        assertEquals(buildSplitSchema(ddlSerializationValue, DEFAULT_TEST_COLUMNS),
+                buildExpectedProperties(expectedDDLSerialization, DEFAULT_TEST_COLUMNS));
+    }
+
+    @Test
+    public void shouldReturnPropertiesWithoutDoubleCommaInColumnsName()
+    {
+        String ddlSerializationValue = "struct article { varchar article, varchar author, date date_pub, int quantity }";
+        String expectedDDLSerialization = "struct article { varchar article, varchar author, date date_pub, int quantity}";
+        assertEquals(buildSplitSchema(ddlSerializationValue, DEFAULT_TEST_COLUMNS),
+                buildExpectedProperties(expectedDDLSerialization, DEFAULT_TEST_COLUMNS));
+    }
+
+    @Test
+    public void shouldOnlyGetColumnTypeFromHiveObjectAndNotFromDDLSerial()
+    {
+        String ddlSerializationValue = "struct article { int article, double author, xxxx date_pub, int quantity }";
+        String expectedDDLSerialization = "struct article { int article, double author, xxxx date_pub, int quantity}";
+        assertEquals(buildSplitSchema(ddlSerializationValue, DEFAULT_TEST_COLUMNS),
+                buildExpectedProperties(expectedDDLSerialization, DEFAULT_TEST_COLUMNS));
+    }
+
+    @Test(expectedExceptions = NullPointerException.class)
+    public void shouldThrowNullPointerExceptionWhenColumnsIsNull()
+    {
+        updateSplitSchema(new Properties(), null);
+    }
+
+    @Test(expectedExceptions = NullPointerException.class)
+    public void shouldThrowNullPointerExceptionWhenSchemaIsNull()
+    {
+        updateSplitSchema(null, EMPTY_LIST);
```

Comment:
nit: You can use `ImmutableList.of()` instead of `EMPTY_LIST` with the additional benefit of getting rid of the unchecked assignment warning.

Reply:
Refactored